persist/fs/commitlog/writer.go

```diff
@@ -191,9 +213,9 @@ func (w *writer) Write(
 		return err
 	}
 
-	if !seen {
-		// Record we have seen this series
-		w.bitset.set(logEntry.Index)
+	if !definitelySeen {
+		// Record we have seen this series at the current seenIdx
+		w.seenCache[seenIdx] = logEntryIndex
```

Comment:
Any reason why we do a separate check here as opposed to just doing this in the `if !definitelySeen {...}` block?